server/service/middleware/endpoint_utils/endpoint_utils.go

```diff
@@ -413,6 +413,11 @@ func MakeDecoder(
 		for _, fp := range fields {
 			field := fp.V
 
+			if headerTagValue, ok := fp.Sf.Tag.Lookup("header"); ok {
+				field.SetString(r.Header.Get(headerTagValue))
+				continue
+			}
```

Comment:
Any chance this will break fleetd on Windows/macOS/Linux? (the `Authentication` header is very common). I would smoke-test it.